hydra-node/src/Hydra/Ledger/Cardano/Orphans.hs

```diff
@@ -0,0 +1,563 @@
+
+safeHashFromText ::
+  (Crypto crypto, MonadFail m) =>
+  Text ->
+  m (Ledger.SafeHash crypto any)
+safeHashFromText t =
+  case Crypto.hashFromTextAsHex t of
+    Nothing -> fail "failed to decode from base16."
+    Just h -> pure $ Ledger.unsafeMakeSafeHash h
+
+--
+-- Script
+--
+
+instance
+  ( ToCBOR (Ledger.Alonzo.Script era)
+  ) =>
+  ToJSON (Ledger.Alonzo.Script era)
+  where
+  toJSON = String . decodeUtf8 . Base16.encode . serialize'
+
+instance
+  ( Crypto (Ledger.Crypto era)
+  , Typeable era
+  ) =>
+  FromJSON (Ledger.Alonzo.Script era)
+  where
+  parseJSON = withText "Script" $ \t ->
+    case Base16.decode $ encodeUtf8 t of
+      Left err -> fail $ "failed to decode from base16: " <> show err
+      Right bs' -> case decodeAnnotator "Script" fromCBOR (fromStrict bs') of
+        Left err -> fail $ show err
+        Right v -> pure v
+
+--
+-- ScriptHash
+--
+
+instance Crypto crypto => ToJSONKey (Ledger.ScriptHash crypto) where
+  toJSONKey = toJSONKeyText $ \(Ledger.ScriptHash h) ->
+    decodeUtf8 $ Base16.encode (Crypto.hashToBytes h)
+
+instance Crypto crypto => FromJSONKey (Ledger.ScriptHash crypto) where
+  fromJSONKey = FromJSONKeyTextParser $ \t ->
+    case Crypto.hashFromTextAsHex t of
+      Nothing -> fail "failed to decode from base16."
+      Just h -> pure $ Ledger.ScriptHash h
+
+--
+-- Timelock
+--
+
+instance ToJSON (Ledger.Mary.Timelock StandardCrypto) where
+  toJSON = String . decodeUtf8 . Base16.encode . serialize'
+
+instance FromJSON (Ledger.Mary.Timelock StandardCrypto) where
+  parseJSON = withText "Timelock" $ \t ->
+    case Base16.decode $ encodeUtf8 t of
+      Left e -> fail $ "failed to decode from base16: " <> show e
+      Right bs' -> case decodeAnnotator "Timelock" fromCBOR (fromStrict bs') of
+        Left err -> fail $ show err
+        Right v -> pure v
+
+--
+-- TxBody
+--
+
+instance ToJSON (Ledger.Alonzo.TxBody StandardAlonzo) where
+  toJSON b =
+    object $
+      mconcat
+        [ onlyIf (const True) "inputs" (Set.map fromShelleyTxIn (Ledger.Alonzo.inputs' b))
+        , onlyIf (not . null) "collateral" (Set.map fromShelleyTxIn (Ledger.Alonzo.collateral' b))
+        , onlyIf (const True) "outputs" (fromShelleyTxOut ShelleyBasedEraAlonzo <$> Ledger.Alonzo.outputs' b)
+        , onlyIf (not . null) "certificates" (Ledger.Alonzo.certs' b)
+        , onlyIf (not . null . Ledger.unWdrl) "withdrawals" (Ledger.Alonzo.wdrls' b)
+        , onlyIf (const True) "fees" (Ledger.Alonzo.txfee' b)
+        , onlyIf (not . isOpenInterval) "validity" (Ledger.Alonzo.vldt' b)
+        , onlyIf (not . null) "requiredSignatures" (Ledger.Alonzo.reqSignerHashes' b)
+        , onlyIf ((/=) mempty) "mint" (fromMaryValue (Ledger.Alonzo.mint' b))
+        , onlyIf isSJust "scriptIntegrityHash" (Ledger.Alonzo.scriptIntegrityHash' b)
+        , onlyIf isSJust "auxiliaryDataHash" (Ledger.Alonzo.adHash' b)
+        , onlyIf isSJust "networkId" (Ledger.Alonzo.txnetworkid' b)
```

Comment:
Flipping the predicate seems to be the more natural case to identify, i.e. when to exclude: `excludeIf null "requiredSignatures" .. `